lib/_http_client.js

```diff
@@ -806,6 +806,9 @@ function onSocketNT(req, socket, err) {
         socket.emit('free');
       } else {
         finished(socket.destroy(err || req[kError]), (er) => {
+          if (er.code === 'ERR_STREAM_PREMATURE_CLOSE') {
```

Comment:
Should we be checking that `er` is set first? (Similarly for the addition in _http_incoming.js)

Reply:
Good catch! Thanks.